### DISCUSSION TITLE
Raw ores & processing chains - 1.17

### DISCUSSION THREAD

**Discussion by GabrielOlvH** [GabrielOlvH/Industrial-Revolution#229] [Ideas]
Due to the new way ores work, gameplay changes are due for every tech mod. This discussion is to define how Industrial Revolution will be affected by these changes. Of course if different community-wide standards are created, indrev will comply.

## Raw ores
This change affects how the infamous early ore doubling since you can't get ore blocks (without Silk Touch) and they are affected by fortune. My idea was to change the ore doubling to be a 1.5x using raw ores instead, which can be achieved two ways: secondary products and small dusts (4 small dusts = 1 dust) or change recipes to be `2 Raw ores = 3 Ore dusts`, I'm leaning more towards the latter.
With this, ore tripling would become `1 Ore block = 3  Ore dusts` which although is less than the max amount you can get from Fortune III, it is constant. I actually would like this to be done in a new machine which would process `1 Ore block` into `2 Raw ores` but I'm not so sure about it.
The old ore tripling and ore quadrupling would be buffed to be quadrupling and quintupling, respectively, and would use ore blocks. The cost for these recipes would be tweaked to be slightly faster and cheaper than the current state

The Mining Rig would continue to provide ore blocks and it would become an essential part of the ore tripling, quadrupling and quintupling chains. Vein sizes will be slightly increased to compensate for ore blocks being "worse" than raw ores and the weight of ores will be increased so you'll see a lot less garbage.

 **None of these changes are set in stone,** if you disagree with any of them or have a better idea please reply.

**Comment by matthewperiut** [comment]
I know this will be a little controversial, but it’s probably the most vanilla-friendly way now considering the new way Mojang intended ores to be its the most obvious and probably will be the most used.

My thoughts are that ore doubling no longer occurs early game with the pulverizer because it wouldn’t make sense as now iron, copper, and gold now act as diamonds, emeralds, and coal do. So keeping ore doubling by ores gotten with silk touch makes sense into multiple two raw pieces of said material, and raw iron becomes iron dust and potentially compressing iron dust becoming raw iron. So it would be raw iron and iron dust that’s exchangeable but iron ingots can’t be turned into dust or raw iron but both iron dust and raw iron can be smelted. This way alternate ways can double ores without making loops 1:1 ratios for the most part just as you would expect as the obvious behaviors because the fundamental idea in ore doubling has been broken down by mojang.

You could make a high level furnace machine that can produce multiple ingots from raw metals, for example a laser machine that finely refines metals unlike a messy furnace, efficiently giving you extra iron at 20% more often or 50% more often but on the other hand using a significant amount of electricity. Maybe a trade-off between efficiency and returns but exponentially more electricity as you get 10% more each time, but the end output is ingots that cannot be turned into dust or raw metal

Removing suggestion from the claim, vanilla friendly mods should be logical and making ore doubling right now wouldn’t be logical, as the whole basis of getting more out of ore doesn’t make sense in the current climate

Although ore doubling is such a good thing that it shouldn’t be removed outright just not how it was because it wouldn’t make sense like that anymore.

I believe other mods will take this approach because it’s the most obvious way, and making it so that you had a 50% chance for more, stuff like that gets really inconsistent from mod to mod making one system better than another and then a machine is never used from one mod because there’s another mod that gives you more and the wiggle room created from these differences will surely create some loopholes down the line

**Reply by GabrielOlvH** [reply]
Iron, copper and gold do not act like diamonds, emeralds and coal. Raw ores are not the final product, they still require processing. Being able to freely convert dust and raw ore back and forth has no use and makes dust useless, might as well remove them. Also dust has always been made with the refined version of the material so I don't see that changing because there's no reason to.

Adding a high level furnace that is worse than other processing chains is basically just removing half of the mod altogether. Why wouldn't ore doubling be logical? I don't understand that.

I don't think other mods will take this approach and remove solid and consistent routes to ones based on luck.

**Comment by Noaaan** [comment]
Ore doubling in 1.17 is challenging, and in an early-game setting it looks to be slightly obsolete, as rushing into mid to late-game where fortune is the go-to enchant might end up being what players do anyway.

I think the focus of using raw ore in mods should move to more interesting methods of processing them. An example of a mod that does this is Mekanism, which adds a scalable and complex method of getting 5x the yield of an ore.

I honestly think with ore doubling dusts are obsolete as an early game option. You should instead now be able to process (I.E crush) an ore block into raw ore itself, saving you a step of the process. Dusts should instead be something gained from a more complex and rewarding process (I.E. a pulverizer chain, maybe requiring a special machine to process). 

To me in an ideal processing chain is crushing ore blocks yielding a consistent amount of raw ore, and pulverizing the raw ore into dust, using that in alloy smelters and/or furnaces.
 
I want to note that once you have processed it (raw ore -> smelted/pulverized) you should not be able to turn it back. Uncooking/uncrafting stuff is concerning.
I do not know how having raw ore yield an 1.5x output over an 2.0x output as dust feels in a gameplay sense, so some experimentation needs to be done in this regard.

**Reply by GabrielOlvH** [reply]
I don't think it'll become obsolete. I believe some mods might split and take different routes but ore doubling will still be a thing widely used because doing it manually takes effort & luck to get double output, while having a machine doing it gives you consistent output and doesn't take much effort, so it's just a trade-off. 

Industrial Revolution already has longer processing chains to get higher ore -> ingot ratio, currently being up to 4x.

Why would ore doubling be obsolete? It's still a fair way of getting a higher ratio from your ores early game where you don't have access to Fortune and Silk touch. Dusts will likely not be touched as they are a fundamental part of the mod (Nikolite and Enriched nikolite dust) and are also a basic material in vanilla (Redstone). 

`To me in an ideal processing chain is crushing ore blocks yielding a consistent amount of raw ore, and pulverizing the raw ore into dust, using that in alloy smelters and/or furnaces.` 
Raw ore doesn't necessarily have to be a part of it but I mostly agree with this. 

And yes, uncrafting will never be a thing for indrev. I'll implement the changes I mentioned and make an alpha release and see how it goes.

**Reply by ghost** [reply]
I would do it just like before just with an extra machine that can crush ore blocks into raw ore and the raw ore in the pulverizer into 2 dust


1 ore block >machine> 1 raw ore>pulverizer>2 dust...
Sorry my english is bad

**Comment by frqnny** [comment]
I personally do not think it is a good idea for the mining rig to get Ore Blocks, it breaks this new and interesting vanilla rule. 
For the "acquiring" raw ores part, you may want to look into perhaps implementing higher quality raw ores that can give you more dust per ore.
You can also expand on the processing- 
https://en.wikipedia.org/wiki/Mineral_processing
imo this is a great article to try to have a realistic ore processing system. Raw Ores now incentivize mods to do stuff more realistically and tech mods should do this kind of stuff. 
Grinding it into ore dust would be the first step. The next step with ore dust would be to either process it into some other type or ore dust- be it "higher quality" or make it produce ore dusts of different sizes, such as a dust of higher granules and one with smaller granules (perhaps smaller granules can be crafted up into higher granules or something?).

Perhaps this ore dust processing system is to complicated and unrelated, but oh well. Point is that mining rig should still give you raw ore & maybe implement some type of higher quality raw ore or something.

**Reply by GabrielOlvH** [reply]
This idea does sound nice but it might cause lots of dupes since I doubt other mods will change the 1 ingot = 1 dust ratio. I have also thought about adding a "Raw ore cluster" for the Mining Rig but then you'd be forced to have a Mining Rig to use the other processing chains which I'm not a fan of since you could use other things like the MI quarry.

**Reply by Noaaan** [reply]
You could simply have it in a way that upgraded machines give you directly more dust when processing raw ore. I agree with frqnny around ore blocks. The expectation for new players with vanilla is that mining up ore blocks grant you raw ore. 
On average fortune 3 gives you 2.2 units of raw iron when you mine an iron ore. A pulverizer could be consistent in this regard, where it always gives you 2. You could then further the process by dusting the raw ore, incentivizing automation.

I want to note that I have not used Indrev, a quick explanation of the current process for increasing ore output would be helpful.

**Reply by GabrielOlvH** [reply]
As of 1.10.0-beta.1:
2 raw ore -> 3 ingots
1 ore block -> 3 ingots
1 ore block in industrial smelter, condenser then pulverizer -> 4 ingots
1 ore block in fluid infuser with sulfuric acid, smelter, condenser then pulverizer -> 5 ingots

Simply "better machine gives more dust" is too broad and I'm not sure how to create actual gameplay for that that isn't just making a new machine for more ores.
Just changing the ore block to raw ore doesn't make sense since the whole idea of the advanced ore processing is "removing the stone bits" to get better results and raw ores don't have stone bits

**Comment by drcrazy** [comment]
Thoughts from Team Reborn:

1. Raw metals is a Mojang approach to ore doubling.  Fortune 3 gives 2.35 raw iron per ore at the cost of durability
2. Raw metals in vanilla are same as ore dusts in modded due to
   - x2-x3 yield requires some special machines \ enchants
   - the only usage is to be smelted into ingots
3. Modded can use more complex approach, more machines, more reagents to get better yield, like x3, x4, x5
4. It looks like metal dusts could be gone and we can use raw metals instead
5. Raw tin, raw silver, raw other ore requires some efforts, especially textures but these more efforts will lead to obvious and clear experience for players. We've got a lot of new players in 1.16. One of the most popular answer during 1.16 was "Please, install REI and check recipe"

Based on thoughts above we are going to have the following in TR:
- modded metal ores will drop raw metal (tin, silver, etc), affected by Fortune
- ore to raw metal grinding recipe. x2 or x3 - haven't decided yet.
- 1 raw metal (tin, silver) will smelt into 1 ingot, like vanilla metals
- 1 ingot (both vanilla and modded) will grind into 1 raw metal, due to raw metal is the new dust 
- metal dusts will be gone

**Comment by ChristopherMeek** [comment]
The only issue I see with this is that if you have an odd number of raw metals then one of them gets "stuck" in the pulveriser which means automation is much harder.  Could the pulveriser only pull multiples of 2 from connected inventories to combat this?

**Reply by GabrielOlvH** [reply]
That's not possible, the only way to solve this issue is to introduce small dusts which isn't something I want to do. When you progress into the mod and get a Mining Rig, it will generate ore blocks so you don't have to deal with this anymore
